compiler/src/lnntoamm/Statement.ts

```diff
@@ -0,0 +1,785 @@
+import { LPNode, NamedAnd, NamedOr } from "../lp";
+import Output from "./Amm";
+import Event from './Event'
+import Fn from "./Fn";
+import opcodes from "./opcodes";
+import Scope from "./Scope";
+import Type, { Builtin, FunctionType } from "./Types";
+import { genName, TODO } from "./util";
+
+export class MetaData {
+  scope: Scope
+  variables: { [name: string]: Dec }
+  retConstraints: Type[]
+
+  constructor(
+    scope: Scope,
+    variables: { [name: string]: Dec } = null,
+    retConstraints: Type[] = null,
+  ) {
+    this.scope = scope;
+    this.variables = variables !== null ? variables : {};
+    this.retConstraints = retConstraints !== null ? retConstraints : [];
+  }
+
+  var(name: string): Dec {
+    if (this.variables[name] == null) {
```

Comment:
`==` instead of `===` maybe changing it explicitly to `this.variables.hasOwnProperty(name)` instead?